src/features/singleRight/delegate/ChooseRightsPage/ChooseRightsPage.tsx

```diff
@@ -1,17 +1,239 @@
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  const [popoverOpen, setPopoverOpen] = useState(false);
+  const [checkedStates, setCheckedStates] = useState<Record<string, boolean>>({});
+  const isSm = useMediaQuery('(max-width: 768px)');
+  const rightsToBeDelegated: DelegationResourceDTO[] = [];
+
+  const chosenServices = useAppSelector((state) => state.singleRightsSlice.chosenServices);
+
+  const delegableChosenServices = chosenServices.filter((s) => s.status !== 'NotDelegable');
+
+  const onRemove = (identifier: string | undefined) => {
+    void dispatch(removeServiceResource(identifier));
+  };
+
+  const onConfirm = () => {
+    setPopoverOpen(!popoverOpen);
+    const filteredList = rightsToBeDelegated.filter((right) => right.checked);
+    const listWithoutChecked = filteredList.map(({ checked, ...rest }) => rest);
+  };
+
+  const handleToggleChecked = (serviceIdentifier: string, rightKey: string) => {
+    setCheckedStates((prevCheckedStates) => ({
+      ...prevCheckedStates,
+      [`${serviceIdentifier}-${rightKey}`]: !prevCheckedStates[`${serviceIdentifier}-${rightKey}`],
+    }));
+  };
+
+  const sortedServiceResources = useMemo(() => {
+    return [...delegableChosenServices]?.sort((a, b) => {
+      const isPartiallyDelegableA = a.status === 'PartiallyDelegable';
+      const isPartiallyDelegableB = b.status === 'PartiallyDelegable';
+
+      if (isPartiallyDelegableA && !isPartiallyDelegableB) {
+        return -1;
+      }
+      if (!isPartiallyDelegableA && isPartiallyDelegableB) {
+        return 1;
+      }
+
+      return a.service?.title.localeCompare(b.service.title);
+    });
+  }, [delegableChosenServices]);
+
+  const serviceResourcesActionBars = useMemo(() => {
+    return sortedServiceResources?.map((chosenService: ChosenService) => {
+      let hasPartiallyDelegableAppeared = false;
+      const isPartiallyDelegable =
+        chosenService.status === 'PartiallyDelegable' && !hasPartiallyDelegableAppeared;
+
+      if (isPartiallyDelegable) {
+        hasPartiallyDelegableAppeared = true;
+      }
+
+      return (
+        <ResourceActionBar
+          key={chosenService.service?.identifier}
+          title={chosenService.service?.title}
+          subtitle={chosenService.service?.resourceOwnerName}
+          status={chosenService.status ?? 'Unchecked'}
+          onRemoveClick={() => {
+            onRemove(chosenService.service?.identifier);
+          }}
+          compact={isSm}
+          canBePartiallyDelegable={true}
+          initialOpen={isPartiallyDelegable}
+        >
+          <div className={classes.serviceResourceContent}>
+            <Paragraph spacing>{chosenService.service?.description}</Paragraph>
+            <Paragraph spacing>{chosenService.service?.rightDescription}</Paragraph>
+            <Paragraph spacing>{t('single_rights.action_bar_adjust_rights_text')}</Paragraph>
+            <Paragraph>{t('single_rights.choose_rights_chip_text')}</Paragraph>
```

Comment:
Thought of the same but if we do that we can't use Paragraph component.

Reply:
No, but you could maybe use the Heading component. Though even on the smallest size (and with h5 or something as heading level) it might be too big. I'm not sure, but I think it's worth a shot 🤷‍♀️

Reply:
I'll try it out 😊